test/qtismtest/runtime/expressions/operators/PatternMatchProcessorTest.php

```diff
@@ -110,7 +110,9 @@ public function patternMatchProvider()
             [new QtiString('string'), 'shell', false],
             [new QtiString('stringString'), '.*', true], // in xml schema 2, dot matches white-spaces
             [new QtiString('^String$'), 'String', false], // No carret nor dollar in xml schema 2
-            [new QtiString('^String$'), '^String$', true],
+            [new QtiString('String'), 'String', true],
+            [new QtiString('String'), '^String$', true],
+            [new QtiString('^String$'), '^String$', false],
             [new QtiString('Str/ing'), 'Str/ing', true],
```

Comment:
Please add tests to match `12.99$` and `$12.99`.

Reply:
Considering $ is usually prepended to a value, I would add just `$12.99` support.